e2e_test.go

```diff
@@ -118,11 +118,25 @@ func runIntegrationTest(t *testing.T, testFile string) {
 			}
 
 			filePath := filepath.Join(dir, tt.Give)
+			if len(tt.WantDiff) > 0 {
+				args := append([]string{"-d", filePath}, args...)
+				var stderr, stdout bytes.Buffer
+				require.NoError(t, run(args, bytes.NewReader(stdin), &stderr, &stdout),
+					"could not run patch")
+				if stdout.String() == "" {
+					// No matches found when applying patch
+					assert.Equal(t, string(tt.WantDiff), stdout.String())
```

Comment:
This assertion will never succeed.
If WantDiff is not empty, then it can't be equal to stdout, which is empty, right?

Reply:
Yeah. I had a different check at the outer loop when I was writing this which would cause errors for indexing [2:] when there was no diff. But the len(wantDiff) >0 check takes care of that. I've removed this redundant check now